### PR TITLE
[NVIDIA TF] Enable BF16 TensorArray scatter, split, unpack

### PR DESCRIPTION
**Pull request by trevor-m**
Support remaining GPU bfloat16 TensorArray ops

### CODE REVIEW

File: tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py [hunk 1]
```diff
@@ -1784,6 +1786,16 @@ def testSkipEagerTensorArrayInt64GPU(self):
       v0, v1 = sess.run([r0, r1], feed_dict={value: [-3, 100]})
       self.assertAllEqual(v0, -3)
       self.assertAllEqual(v1, 100)
+  
+  @test_util.deprecated_graph_mode_only
+  def testTensorArrayScatterBfloat61GPU(self):
```
Comment (reedwm) [inline tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py:1791]: change 61 to 16.

File: tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py [hunk 2]
```diff
@@ -1784,6 +1786,16 @@ def testSkipEagerTensorArrayInt64GPU(self):
...
+    if not test.is_gpu_available():
+      return
+    with self.session(force_gpu=True) as sess:
+      ta = tensor_array_ops.TensorArray(
+          dtype=dtypes.float32, tensor_array_name="foo", size=5)
```
Comment (reedwm) [inline tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py:1796]: Use bfloat16 here

**Review by reedwm — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
Hi @trevor-m Can you please check @reedwm's comments and keep us posted ? Thank you!

File: tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py (hunk 1 above)
Comment (trevor-m) [reply tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py]: Oops, fixed.

File: tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py (hunk 2 above)
Comment (trevor-m) [reply tensorflow/python/kernel_tests/data_structures/tensor_array_ops_test.py]: Thanks @reedwm for the review. Fixed this.

**Review by reedwm — APPROVED** [review]

**Pull request merged**